contracts/cw-proposal-single/src/contract.rs

```diff
@@ -352,18 +346,18 @@ pub fn execute_close(
 ) -> Result<Response, ContractError> {
     let mut prop = PROPOSALS.load(deps.storage, proposal_id)?;
 
-    // Only open and expired or rejected proposals may be closed.
-    match prop.status {
-        Status::Rejected => (),
-        Status::Open => {
-            if !prop.expiration.is_expired(&env.block) || prop.is_passed(&env.block) {
-                return Err(ContractError::WrongCloseStatus {});
-            }
-        }
-        _ => return Err(ContractError::WrongCloseStatus {}),
+    // Update status to ensure that proposals which were open and have expired are moved to "rejected."
+    prop.update_status(&env.block);
+    if prop.status != Status::Rejected {
+        return Err(ContractError::WrongCloseStatus {});
     }
+
     let old_status = prop.status;
-    let refund_message = get_return_deposit_msg(&prop)?;
+
+    let refund_message = match &prop.deposit_info {
+        Some(deposit_info) => get_return_deposit_msg(&deposit_info, &prop.proposer)?,
```

Comment:
Need to check that we are actually refunding failed proposals.
```suggestion
        Some(deposit_info) => if deposit_info.refund_failed_proposals {
            get_return_deposit_msg(&deposit_info, &prop.proposer)?
        } else {
            vec![]
        },
```